### PR TITLE
SUBMARINE-714. Store experiment metadata into mysql database

### PR DESCRIPTION
**Pull request by ByronHsu**
### What is this PR for?
In the past, we temporarily stored experiment metadata in memory. However, it is not persistent, when we restart the server, the data will be lost. Alternatively, we should store it in MySQL database.

### What type of PR is it?
[Improvement]

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-714

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-server/server-core/src/main/java/org/apache/submarine/server/SubmitterManager.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
     LOG.info("Submarine Home: {}", home);
     SUBMITTER_CONFIG_MAP.put("k8s", new JobSubmitterConfig(
         "org.apache.submarine.server.submitter.k8s.K8sSubmitter",
-        home + "/lib/submitter/k8s/"));
+        "/Users/byronhsu/Desktop/submarine/submarine-server/server-submitter/submitter-k8s/target"));
```
Comment (jojochuang) [inline submarine-server/server-core/src/main/java/org/apache/submarine/server/SubmitterManager.java:55]: let's not hardcode the path?

Comment (ByronHsu) [reply submarine-server/server-core/src/main/java/org/apache/submarine/server/SubmitterManager.java:55]: @jojochuang  Thanks! It is a temporary commit, and I accidentally pushed it. There are still bugs in this PR. I am working on fixing it right now.

File: submarine-server/server-core/src/test/java/org/apache/submarine/server/experiment/database/ExperimentServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+      experimentService.insert(entity);
+
+      ExperimentEntity entitySelected = experimentService.select(id);
+
+      compareEntity(entity, entitySelected);
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new Exception(e);
+    }
```
Comment (jojochuang) [inline submarine-server/server-core/src/test/java/org/apache/submarine/server/experiment/database/ExperimentServiceTest.java:61]: No need to try ... catch in a test. Just let it throw exceptions.

File: submarine-server/server-core/src/test/java/org/apache/submarine/server/experiment/database/ExperimentServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        entities.add(entity);
+      }
+
+      List<ExperimentEntity> entities_selected = experimentService.selectAll();
+
+      Assert.assertEquals(SIZE, entities_selected.size());
+      for (int i = 0; i < entities_selected.size(); i++) {
+        compareEntity(entities.get(i), entities_selected.get(i));
+      }
+    } catch (Exception e) {
```
Comment (jojochuang) [inline submarine-server/server-core/src/test/java/org/apache/submarine/server/experiment/database/ExperimentServiceTest.java:84]: ditto

File: submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/database/ExperimentService.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      sqlSession.commit();
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new SubmarineRuntimeException("Unable to get experiment entities from database");
+    }
+    return entities;
+  }
+
+  public ExperimentEntity select(String id) throws SubmarineRuntimeException {
+    LOG.info("Experiment select");
```
Comment (jojochuang) [inline submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/database/ExperimentService.java:49]: log the id too.

File: submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/database/ExperimentService.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      sqlSession.commit();
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new SubmarineRuntimeException("Unable to update experiment entity in database");
+    }
+    return true;
+  }
+
+  public boolean delete(String id) throws SubmarineRuntimeException {
+    LOG.info("Experiment delete");
```
Comment (jojochuang) [inline submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/database/ExperimentService.java:89]: log the id too.

File: submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/database/ExperimentEntity.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+  public ExperimentEntity() {}
+
+  public String getExperimentSpec() {
+    return experimentSpec;
+  }
+
+  public void setExperimentSpec(String experimentSpec) {
+    this.experimentSpec = experimentSpec;
+  }
```
Comment (jojochuang) [inline submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/database/ExperimentEntity.java:38]: add a toString() method so we can log this object.

File: submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/database/ExperimentService.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      sqlSession.commit();
+    } catch (Exception e) {
+      LOG.error(e.getMessage(), e);
+      throw new SubmarineRuntimeException("Unable to get experiment entity from database");
+    }
+    return entity;
+  }
+
+  public boolean insert(ExperimentEntity experiment) throws SubmarineRuntimeException {
+    LOG.info("Experiment insert");
```
Comment (jojochuang) [inline submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/database/ExperimentService.java:63]: we can consider log the ExperimentEntity object. (if it's too much, log it at debug log level)

**Review by jojochuang — CHANGES_REQUESTED** [review]

**Review by pingsutw — COMMENTED** [review]
LGTM, Thanks @ByronHsu for the contribution, and thanks @jojochuang for the review.

**Comment by pingsutw** [comment]
btw, the test failure is unrelated to this PR. Test passed in https://travis-ci.com/github/apache/submarine/builds/214343196

**Pull request closed without merging**